.github/workflows/release-promotion.yml

```diff
@@ -0,0 +1,53 @@
+name: Promote Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment'
+        required: true
+        type: choice
+        options:
+          - staging
+          - production
+      version:
+        description: 'Product version'
+        required: true
+        type: string
+      sha:
+        description: 'SHA'
+        required: true
+        type: string
+      branch:
+        description: 'BRANCH'
+        required: true
+        type: string
+
+jobs:
+  promote:
+    name: "Promote release"
+    runs-on: [self-hosted, linux]
+    steps:
+      - name: Setup bob
+        uses: hashicorp/action-setup-bob@v1
+        with:
+          github-token:
+            ${{ secrets.ELEVATED_GITHUB_TOKEN}}
+      - name: Promote to ${{ inputs.environment }}
+        env:
+          BOB_GITHUB_TOKEN: ${{ secrets.ELEVATED_GITHUB_TOKEN }}
+          VERSION: ${{ github.event.inputs.version }}
+          SHA: ${{ github.event.inputs.sha }}
+          BRANCH: ${{ github.event.inputs.branch }}
+          ENVIRONMENT: ${{ github.event.inputs.environment }}
+        run: |
+          bob trigger-promotion \
```

Comment:
🥳 very excited for this! Last time we discussed it there were concerns about secrets management for having this workflow in a public repo - I assume that's not something we still need to worry about?

Reply:
I don't think this is a concern given that we can tightly control the token's permissions now by using fine grained tokens. I specified `secrets.ELEVATED_GITHUB_TOKEN` just as an example as I saw it existed in the repo. But we would want to generate a new finely grained token (as the terraform bot user) and give it access to the `hashicorp/crt-workflows-common` repo and add Actions (Read and write) as the repo permissions. This would have to be done by someone on the Terraform team with access to log in as this user.